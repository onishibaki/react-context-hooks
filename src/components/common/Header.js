import React from "react";
import styled from "styled-components";

const Unordered = styled.ul`
  list-style: none;
  padding: 0;
  border: 1px solid #000000;
  display: flex;
  justify-content: center;
  justify-content: space-around;
  align-items: center;
  height: 350px;
  a {
    text-decoration: none;
    border: 1px solid black;
    padding: 20px 60px;
    font-weight: 700;
    color: black;
    &:hover {
      background: #00000059;
      color: white;
    }
  }
`;

const Header = () => {
  return (
    <header className="main-header">
      <nav>
        <Unordered className="something">
          <li>
            <a href="/admin">Admin</a>
          </li>
          <li>
            <a href="/employee">Employee</a>
          </li>
        </Unordered>
      </nav>
    </header>
  );
};

export default Header;
