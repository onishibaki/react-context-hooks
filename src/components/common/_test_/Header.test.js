import React from "react";
import { render, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import Header from "../Header";

afterEach(cleanup);

describe("Header Test", () => {
  it("Renders Correctly", () => {
    const div = document.createElement("div");
    render(<Header />, div);
  });

  it("Matches Snapshot", () => {
    const headerSnapshot = renderer.create(<Header />).toJSON();
    expect(headerSnapshot).toMatchSnapshot();
  });
});
