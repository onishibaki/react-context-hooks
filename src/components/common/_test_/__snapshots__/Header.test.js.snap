// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Test Matches Snapshot 1`] = `
<header
  className="main-header"
>
  <nav>
    <ul
      className="sc-AxjAm fPMbNd something"
    >
      <li>
        <a
          href="/admin"
        >
          Admin
        </a>
      </li>
      <li>
        <a
          href="/employee"
        >
          Employee
        </a>
      </li>
    </ul>
  </nav>
</header>
`;

exports[`Headers Test Matches Snapshot 1`] = `
<header
  className="main-header"
>
  <nav>
    <ul
      className="sc-AxjAm fPMbNd something"
    >
      <li>
        <a
          href="/admin"
        >
          Admin
        </a>
      </li>
      <li>
        <a
          href="/employee"
        >
          Employee
        </a>
      </li>
    </ul>
  </nav>
</header>
`;

exports[`Matches Snapshot 1`] = `
<header
  className="main-header"
>
  <nav>
    <ul
      className="sc-AxjAm fPMbNd something"
    >
      <li>
        <a
          href="/admin"
        >
          Admin
        </a>
      </li>
      <li>
        <a
          href="/employee"
        >
          Employee
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
